### PR TITLE
feat:upgraded the app with treasury module to SDK v0.50

### PR DESCRIPTION
**Pull request by vishal-kanna**

### CODE REVIEW

**Comment by aleem1314** [comment]
Check #1036 PR review comments. Apply relevant changes here too.

**Review by aleem1314 — CHANGES_REQUESTED** [review]
Still there are few things need to be updated.
Check https://github.com/palomachain/paloma/pull/1036#pullrequestreview-1763965986

**Comment by aleem1314** [comment]
@vishal-kanna resolve conflicts.

File: x/treasury/types/store.go [hunk 1]
```diff
@@ -1,19 +1,25 @@
...
 type Store struct {
-	StoreKey storetypes.StoreKey
+	StoreKey cosmosstore.KVStoreService
 }
 
-func (sg Store) TreasuryStore(ctx sdk.Context) sdk.KVStore {
-	return prefix.NewStore(ctx.KVStore(sg.StoreKey), []byte("treasury"))
+func (sg Store) TreasuryStore(ctx context.Context) storetypes.KVStore {
+	store := runtime.KVStoreAdapter(sg.StoreKey.OpenKVStore(ctx))
+	return prefix.NewStore(store, []byte("treasury"))
```
Comment (aleem1314) [inline x/treasury/types/store.go:24]:
```suggestion
	return runtime.KVStoreAdapter(sg.StoreKey.OpenKVStore(ctx))
```

Comment (vishal-kanna) [reply x/treasury/types/store.go:24]: Ok

**Review by byte-bandit — DISMISSED** [review]
lgtm, final review goes to @aleem1314

File: x/treasury/genesis.go [hunk 2]
```diff
@@ -1,33 +1,38 @@
...
 	"github.com/palomachain/paloma/x/treasury/keeper"
 	"github.com/palomachain/paloma/x/treasury/types"
 )
 
 // InitGenesis initializes the capability module's state from a provided genesis
 // state.
-func InitGenesis(ctx sdk.Context, k keeper.Keeper, genState types.GenesisState) {
-	k.SetParams(ctx, genState.Params)
+func InitGenesis(ctx context.Context, k keeper.Keeper, genState types.GenesisState) {
+	Ctx := sdk.UnwrapSDKContext(ctx)
```
Comment (aleem1314) [inline x/treasury/genesis.go:14]:
```suggestion
	sdkCtx := sdk.UnwrapSDKContext(ctx)
```

go variables must start with small letter.

File: x/treasury/genesis.go [hunk 3]
```diff
@@ -1,33 +1,38 @@
...
+	err = k.SetSecurityFee(Ctx, "0.01")
 	if err != nil {
 		panic(err)
 	}
 }
 
 // ExportGenesis returns the capability module's exported genesis.
-func ExportGenesis(ctx sdk.Context, k keeper.Keeper) *types.GenesisState {
+func ExportGenesis(ctx context.Context, k keeper.Keeper) *types.GenesisState {
+	Ctx := sdk.UnwrapSDKContext(ctx)
```
Comment (aleem1314) [inline x/treasury/genesis.go:30]:
```suggestion
	sdkCtx := sdk.UnwrapSDKContext(ctx)
```

File: x/treasury/genesis.go (hunk 2 above)
Comment (vishal-kanna) [reply x/treasury/genesis.go:14]: ok,I'll update it

**Review by aleem1314 — APPROVED** [review]
lgtm

**Review by aleem1314 — APPROVED** [review]
lgtm

**Review by taariq — APPROVED** [review]
LGTM!

**Pull request merged**